src/gateways/accountHolderApiGateway.ts

```diff
@@ -0,0 +1,25 @@
+import axios, { AxiosResponse } from "axios";
+import { IAccountHolderIdResponseBody } from "../lib/accountHolder/accountHolderIdResponseBody";
+
+export class AccountHolderApiGateway {
+  private readonly apiUrl: string;
+  private readonly accountHolderIdEndpoint = "account-holder/auth-id";
+
+  constructor() {
+    this.apiUrl = process.env.API_URL;
+  }
+
+  public async getAccountHolderId(
+    authId: string,
+    accessToken: string
+  ): Promise<string> {
+    const url = `${this.apiUrl}/${this.accountHolderIdEndpoint}/${authId}`;
+    let response = await axios.get<
+      any,
+      AxiosResponse<IAccountHolderIdResponseBody>
+    >(url, {
+      headers: { Authorization: `Bearer ${accessToken}` },
+    });
```

Comment:
Second bite of the cherry!  Sorry.

Do we want to test that Axios' exceptions bubble up?  I'd say "yes" because it's a really cheap/easy test to write and we want confidence that exceptions make their way up to the use case that will use this gateway (e.g. if `404` account holder not found is returned).

We shouldn't rely on our current knowledge of how Axios works to provide us with that confidence because:

- Future us / other devs might not have that same knowledge.
- Even with that knowledge, it doesn't mean future us / other devs won't break it and not notice (without a test).
- Axios is an implementation detail and we shouldn't test/decide not to test because of implementation details.

Reply:
done :)